src/signal.rs

```diff
@@ -26,34 +27,54 @@ fn notify(signals: &[i32]) -> channel::Receiver<i32> {
 }
 
 impl Waiter {
-    pub fn start() -> Waiter {
-        Waiter {
-            receiver: notify(&[
-                SIGINT, SIGTERM,
-                SIGUSR1, // allow external triggering (e.g. via bitcoind `blocknotify`)
-            ]),
-        }
-    }
+    pub fn start() -> (Sender<BlockHash>, Waiter) {
+        let (block_hash_notify, block_hash_receive) = channel::bounded(1);
```

Comment:
It feels more natural to me to create the `Sender` outside and accept it as an argument, but I don't feel strongly about this

Reply:
4edd741592ec00c3f1e33969ed26fc9f395f90e0